challenges/08-coding-interview-prep/rosetta-code.json

```diff
@@ -10506,6 +10506,1096 @@
+        }
+      ],
+      "id": "5a23c84252665b21eecc7f3f",
+      "challengeType": 5,
+      "releasedOn": "July 28, 2018",
+      "files": {
+        "indexjs": {
+          "key": "indexjs",
+          "ext": "js",
+          "name": "index",
+          "contents": ["function numberName (value) {", "  // Good luck!", "}"],
+          "head": [],
+          "tail": []
+        }
+      }
+    },
+    {
+      "title": "Numeric error propagation",
+      "description": [
+        "If  <b>f</b>,  <b>a</b>,  and  <b>b</b>  are values with uncertainties  σ<sub>f</sub>,  σ<sub>a</sub>,  and  σ<sub>b</sub>,  and  <b>c</b>  is a constant; then if  <b>f</b>  is derived from  <b>a</b>,  <b>b</b>,  and  <b>c</b>  in the following ways, then  σ<sub>f</sub>  can be calculated as follows:",
+        "Addition/Subtraction",
+        "If  f = a &plusmn; c,  or  f = c &plusmn; a  then  <b>σ<sub>f</sub> = σ<sub>a</sub></b>",
+        "If  f = a &plusmn; b  then  <b>σ<sub>f</sub><sup>2</sup> = σ<sub>a</sub><sup>2</sup> + σ<sub>b</sub><sup>2</sup></b>",
+        "Multiplication/Division",
+        "If  f = ca  or  f = ac    then  <b>σ<sub>f</sub> = |cσ<sub>a</sub>|</b>",
+        "If  f = ab  or  f = a / b             then  <b>σ<sub>f</sub><sup>2</sup> = f<sup>2</sup>( (σ<sub>a</sub> / a)<sup>2</sup> + (σ<sub>b</sub> / b)<sup>2</sup>)</b>",
+        "Exponentiation",
+        "If  f = a<sup>c</sup>  then  <b>σ<sub>f</sub> = |fc(σ<sub>a</sub> / a)|</b>",
+        "Caution:",
+        "This implementation of error propagation does not address issues of dependent and independent values.  It is assumed that  <b>a</b>  and  <b>b</b>  are independent and so the formula for multiplication should not be applied to  <b>a*a</b>  for example.  See  <a href=\"http://rosettacode.org/wiki/Talk:Numeric_error_propagation\">the talk page</a>  for some of the implications of this issue.",
+        "<ol><li>Add an uncertain number type to your language that can support addition, subtraction, multiplication, division, and exponentiation between numbers with an associated error term together with 'normal' floating point numbers without an associated error term. Implement enough functionality to perform the following calculations.</li> <li>Calculate the distance between the two points using the classic Pythagorean formula:   d = &radic;<span style=\"text-decoration:overline\">  (x1 - x2)²  +  (y1 - y2)²  </span> </li></ol>",
+        "Write a function that takes x1, x2, y1 and y2 as parameters. These parameters are arrays having 2 elements. The first is the value and the second is the error. The function should return the result after the above calculation as an object in this form { \"value\" : val, \"error\" : err }. Both value and error should be rounded to 4 decimal places."
```

Comment:
`"Write a function that takes x1, x2, y1, and y2 as parameters. These parameters are arrays with 2 elements. The first is the value and the second is the error. The function should return the result after the above calculation as an object in this form { \"value\" : val, \"error\" : err }. Both the value and error should be rounded to 4 decimal places."`